### PR TITLE
Make InternalSDKUtils AOT compatible in V4

### LINKED ISSUE #3497: InternalSDKUtils trim warning
### Describe the bug

Usage of AWSSDK.Core 3.7.400.22 with native aot causes the following trimmer warning.

ILC : Trim analysis warning IL2075: Amazon.Util.Internal.InternalSDKUtils.ApplyValues(Object,IDictionary`2<String,Object>): 'this' argument does not satisfy 'DynamicallyAccessedMemberTypes.PublicProperties' in call to 'System.Type.GetProperty(String)'. The return value of method 'System.Object.GetType()' does not have matching annotations. The source value must declare at least the same requirements as those declared on the target location it is assigned to.

### Regression Issue

- [ ] Select this option if this issue appears to be a regression.

### Expected Behavior

No trim warnings.

### Current Behavior

Trim warnings raised.

### Reproduction Steps

Reference AWSSDK.Core and try to publish with PublishAot true...  Various parts of my csproj are below.

```
  <PropertyGroup>
    <TrimmerSingleWarn>false</TrimmerSingleWarn> 
  </PropertyGroup>

  <ItemGroup> 
    <PackageReference Include="Amazon.Lambda.RuntimeSupport" Version="1.11.0" />
    <PackageReference Include="Amazon.Lambda.Core" Version="2.3.0" />
    <PackageReference Include="Amazon.Lambda.Serialization.SystemTextJson" Version="2.4.3" />
    <PackageReference Include="Amazon.Lambda.APIGatewayEvents" Version="2.7.1" />
  </ItemGroup>

  <ItemGroup>
    <!-- <TrimmerRootAssembly Include="AWSSDK.Core" /> -->
    <TrimmerRootAssembly Include="Amazon.Lambda.APIGatewayEvents" />
    <TrimmerRootAssembly Include="bootstrap" />
  </ItemGroup>
```

### Possible Solution

_No response_

### Additional Information/Context

Also, setting the AWSSDK.Core reference as a trimmer root assembly results in even more warnings.

### AWS .NET SDK and/or Package version used

AWSSDK.Core 3.7.400.22

### Targeted .NET Platform

.NET 8

### Operating System and version

OSX Arm64

### PR DESCRIPTION
**Pull request by peterrsongg**
## Description

In AWSSDK.Core 3.7.400.35 `InternalSDKUtils` was made AOT compatible by obsoleting `InternalSDKUtils.ApplyValues` and creating a new method `InternalSDKUtils.ApplyValuesV2`, which is AOT compatible. This PR removes the obsolete method in V4 and refactors existing methods to use V2 similar to in V3. Thanks @Dreamescaper for the suggestion!  

## Motivation and Context

Fixes #3497  in V4
## Testing

Created a console app with <PublishTrimmed> set to true and published the console app and no warnings appeared. I even ran the .exe and it worked.

DRY_RUN passes

## Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist

- [ ] My code follows the code style of this project
- [ ] My change requires a change to the documentation
- [ ] I have updated the documentation accordingly
- [ ] I have read the **README** document
- [ ] I have added tests to cover my changes
- [x] All new and existing tests passed

## License

- [x] I confirm that this pull request can be released under the Apache 2 license

[issues]: https://github.com/aws/aws-sdk-net/issues
[license]: http://aws.amazon.com/apache2.0/
[cla]: http://en.wikipedia.org/wiki/Contributor_License_Agreement

### CODE REVIEW

**Review by normj — APPROVED** [review]

File: sdk/src/generator/.DevConfigs/c9da7660-10c9-4a8a-b5b4-5fdffe5d9642.json
Comment (dscpinheiro) [inline sdk/src/generator/.DevConfigs/c9da7660-10c9-4a8a-b5b4-5fdffe5d9642.json:1]: This file shouldn't be here. Copy and paste error maybe?

Comment (peterrsongg) [reply sdk/src/generator/.DevConfigs/c9da7660-10c9-4a8a-b5b4-5fdffe5d9642.json:1]: good call, removed!

Comment (peterrsongg) [reply sdk/src/generator/.DevConfigs/c9da7660-10c9-4a8a-b5b4-5fdffe5d9642.json:1]: added it back in the correct location

Comment (dscpinheiro) [reply sdk/src/generator/.DevConfigs/c9da7660-10c9-4a8a-b5b4-5fdffe5d9642.json:1]: Did you push your changes? I still see the file in the PR.

Comment (peterrsongg) [reply sdk/src/generator/.DevConfigs/c9da7660-10c9-4a8a-b5b4-5fdffe5d9642.json:1]: did you want me to remove it completely? I just placed it in the correct path

**Review by dscpinheiro — APPROVED** [review]

**Pull request merged**